src/main/java/proxy/ImageLoader.java

```diff
@@ -0,0 +1,27 @@
+package proxy;
+
+
+import proxy.cache.IImageProvider;
+
+import java.util.HashMap;
+
+public class ImageLoader {
+    private IImageProvider api;
+    HashMap<String, Image> list;
+    public ImageLoader(IImageProvider imageProvider) {
+        this.api = imageProvider;
+    }
+
+    public String imageValidator(User user){
```

Comment:
I think the purpose is to provide images, not validate, validation is a hidden feature of this class. the first thing that I think when I see an `ImageLoader` class is to expect a `load` method, not a   `validator` method, food for thought